go/controller/api/v1alpha1/autogenagent_types.go

```diff
@@ -35,6 +35,8 @@ type AgentSpec struct {
 	ModelConfig string `json:"modelConfig,omitempty"`
 	// +kubebuilder:validation:MaxItems=20
 	Tools []*Tool `json:"tools,omitempty"`
+	// +optional
+	Memory []*Memory `json:"memory,omitempty"`
```

Comment:
should this be a list of references to Memory objects? it looks like we are inlining the whole Memory crd inside of the agent CRD here

Reply:
Agreed, there is a new CRD referenced below, but then you inline it here. I'm cool with supporting inline as well as references, but that's not quite what's happening here.

Reply:
yes, should be a list not sure why i inlined it